src/twisted/conch/ssh/filetransfer.py

```diff
@@ -498,6 +498,13 @@ def connectionLost(self, reason):
         """
         Clean all opened files and directories.
         """
+
+        # On Python2, we're an oldstyle class, so we can't use super().
+        FileTransferBase.connectionLost(self, reason)
+        # Not sure why self.connected isn't updated in
```

Comment:
`IProtocol` is the interface that the *transport* requires of the protocol, so it's deliberately minimal.  The commonality of attributes like `connecting`, `disconnected`, et. al., are mostly an accident of history.

Reply:
`IProtocol` doesn't, but `t.i.p.BaseProtocol` does have a `connected` attribute and updates it in some situations, just not others; that's why it seems a little odd to me that `t.i.p.Protocol` (the first ancestor class that implements `connectionLost()`) doesn't update `connected`. Or, perhaps, since `Protocol` has no method bodies, its method docstring should suggest that implementers update `connected`? Or `BaseProtocol` should lose the `connected` attribute and it should be implemented in each subclass that wants it? This seems like a question for a separate PR. :)

In either case, I think you're right I should move that code up into `FileTransferBase`.